### ISSUE TITLE
vglrun error: X Error of failed request:  BadMatch (invalid parameter attributes)

### ISSUE THREAD

**Issue by chekhovana** [VirtualGL/virtualgl#206]
I am trying to launch remote OpenGL application using TurboVNC + VirtualGL. glxinfo and glxsphere run successfully, but my app launch results in error:

```
X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  130 (MIT-SHM)
  Minor opcode of failed request:  3 (X_ShmPutImage)
  Serial number of failed request:  40
  Current serial number in output stream:  41
```

I wrote minimal python example which reproduces the problem:

```
import glfw
if not glfw.init():
    print("Cannot initialize GLFW")
    exit()
window = glfw.create_window(320, 240, "OpenGL window", None, None)
```

**Comment by chekhovana** [comment]
The full procedure is as follows:

**1. [local machine] Connect to server:**

`/opt/VirtualGL/bin/vglconnect -s user@ipaddress`

**2. [remote server] Launch vncserver:**

`vncserver -wm xfce`

**3. [remote server] Check if server is running:**

`vncserver -list`

The output is:

```
TurboVNC sessions:
X DISPLAY #	PROCESS ID	NOVNC PROCESS ID
:1		17657
```

**4. [local machine] Set up ssh tunneling:**

`ssh -L 5901:localhost:5901 -i ~/.ssh/id_rsa user@ipaddress`

**5. [local machine] Launch vncviewer:**

`/opt/TurboVNC/bin/vncviewer`

Enter server address:
<img width="503" alt="vncviewer" src="https://user-images.githubusercontent.com/6779698/174225106-37d713bc-8e4b-4aaa-a042-34336ede7a19.png">
Connection is successfully established, I can see the the following window:
<img width="1262" alt="Screenshot 2022-06-17 at 08 09 09" src="https://user-images.githubusercontent.com/6779698/174224055-db7facab-6990-4d11-b798-9c2fd9c2e2c7.png">
There is some problem with rendering (black rectangle in upper left corner).

**6. [remote machine] Run minimal python example:**

`vglrun -d :1 python3 example.py`

example.py contains the following code:

```
import glfw
if not glfw.init():
    print("Cannot initialize GLFW")
    exit()
window = glfw.create_window(320, 240, "OpenGL window", None, None)
```

The output is:

```
X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  130 (MIT-SHM)
  Minor opcode of failed request:  3 (X_ShmPutImage)
  Serial number of failed request:  39
  Current serial number in output stream:  40
```

**Comment by dcommander** [comment]
You are fundamentally misunderstanding the usage of VirtualGL and TurboVNC.  Please refer to the User's Guides.  You previously said that you do not want a remote desktop.  Therefore you do not need TurboVNC.  You simply need to use `vglconnect` to connect to the server and `vglrun` to run the OpenGL application.  However, you are also misunderstanding the purpose of the `vglrun -d` argument.  `vglrun -d` or `VGL_DISPLAY` are used to specify a 3D X server display.  A 3D X server is a GPU-attached X server running on the server machine, usually listening on Display :0.0 (which is the default value of `VGL_DISPLAY`.)  You should run `vglserver_config` as root on the server machine to grant access to the 3D X server while it is sitting at the login prompt.  (Again, this is all in the User's Guide.)  Then VirtualGL can redirect OpenGL rendering from an OpenGL application to the 3D X server, read back the rendered images, and transport them to the 2D X server.  The 2D X server is either an X proxy such as TurboVNC, or it is the X server on your client machine.  Since you opted not to use TurboVNC, then the 2D X server is XQuartz in your case.  When using VirtualGL, the 2D X server does not do any OpenGL rendering.  It simply receives images of the pre-rendered OpenGL frames from VirtualGL, and the frames are rendered by the 3D X server.

tl;dr: Either use `vglconnect` or TurboVNC, not both.

**Comment by chekhovana** [comment]
I am sorry for being so incompetent in my questions. I admit the subject is very difficult for me due to lack of experience.

Just to summarize: 

I finally succeded with vglconnect without vncserver. The main problem was that, following [Headless nVidia Mini How-To](https://virtualgl.org/Documentation/HeadlessNV), I didn't remove the row

`Option "UseDisplayDevice" "none" `

from xorg.conf. Without this step display manager on remote server didn't start and any further steps failed.

Thank you for cooperation.
